tls/extensions/s2n_extension_type_lists.c

```diff
@@ -0,0 +1,139 @@
+        &s2n_client_sct_list_extension,
+        &s2n_client_max_frag_len_extension,
+        &s2n_client_session_ticket_extension,
+        &s2n_client_supported_groups_extension,
+        &s2n_client_ec_point_format_extension,
+        &s2n_client_pq_kem_extension,
+        &s2n_client_renegotiation_info_extension,
+};
+
+static const s2n_extension_type *const tls12_server_hello_extensions[] = {
+        &s2n_server_supported_versions_extension,
+        &s2n_server_server_name_extension,
+        &s2n_server_ec_point_format_extension,
+        &s2n_server_renegotiation_info_extension,
+        &s2n_server_alpn_extension,
+        &s2n_server_status_request_extension,
+        &s2n_server_sct_list_extension,
+        &s2n_server_max_fragment_length_extension,
+        &s2n_server_session_ticket_extension,
+};
+
+static const s2n_extension_type *const tls13_server_hello_extensions[] = {
+        &s2n_server_supported_versions_extension,
+        &s2n_server_key_share_extension,
+        &s2n_server_cookie_extension,
+};
+
+static const s2n_extension_type *const encrypted_extensions[] = {
+        &s2n_server_server_name_extension,
+        &s2n_server_max_fragment_length_extension,
+        &s2n_server_alpn_extension,
+};
+
+static const s2n_extension_type *const cert_req_extensions[] = {
+        &s2n_server_signature_algorithms_extension,
+};
+
+static const s2n_extension_type *const certificate_extensions[] = {
+        &s2n_tls13_server_status_request_extension,
+        &s2n_server_sct_list_extension,
+};
+
+#define S2N_EXTENSION_LIST(list) { .extension_types = (list), .count = s2n_array_len(list) }
+
+static const s2n_extension_type_list empty_extension_type_list = { .extension_types = NULL, .count = 0 };
+
+static struct {
+    const s2n_extension_type_list tls12_list;
+    const s2n_extension_type_list tls13_list; /* TLS1.3 moved some extensions to different extension lists */
+} extension_lists[] = {
+        [S2N_EXTENSION_LIST_CLIENT_HELLO] = {
+                .tls12_list = S2N_EXTENSION_LIST(client_hello_extensions),
+                .tls13_list = S2N_EXTENSION_LIST(client_hello_extensions),
+        },
+        [S2N_EXTENSION_LIST_SERVER_HELLO] = {
+                .tls12_list = S2N_EXTENSION_LIST(tls12_server_hello_extensions),
+                .tls13_list = S2N_EXTENSION_LIST(tls13_server_hello_extensions),
+        },
+        [S2N_EXTENSION_LIST_ENCRYPTED_EXTENSIONS] = {
+                .tls13_list = S2N_EXTENSION_LIST(encrypted_extensions),
+        },
+        [S2N_EXTENSION_LIST_CERT_REQ] = {
+                .tls13_list = S2N_EXTENSION_LIST(cert_req_extensions),
+        },
+        [S2N_EXTENSION_LIST_CERTIFICATE] = {
+                .tls13_list = S2N_EXTENSION_LIST(certificate_extensions),
+        },
+};
+
+int s2n_extension_type_list_get(s2n_extension_list_id list_type, struct s2n_connection *conn,
+        const s2n_extension_type_list **extension_list)
+{
+    notnull_check(extension_list);
```

Comment:
don't you need a `notnull_check(conn)` as well?

Reply:
`conn` is never de-referenced in this function, just passed on to `s2n_connection_get_protocol_version`.